### DISCUSSION TITLE
How to correctly use WebWorkers

### DISCUSSION THREAD

**Discussion by givip** [simolus3/drift#1385] [Q&A]
I read guide here https://moor.simonbinder.eu/web/#using-web-workers
But unfortunately cannot make it work.
I'm getting exception in browser `Failed to fetch a worker script.`

Why `worker.dart` file should be places in `./web` directory, I thought, all `dart` files should be compiled?

What I'm doing wrong?

**Comment by simolus3** [comment]
Ah, sorry. The web worker example assumes using `build_runner` and `build_web_compilers` to compile your app. Flutter users will probably need to manually compile the web worker entrypoint before making it available. I'll change the documentation to point this out.

But basically yes, all `.dart` files should be compiled to JS so that they can be used. So you'd have a separate `main` function for the service worker which you'd compile somehow (I don't know how that works with Flutter, surely it will be possible). Then you can use that js file as a worker.
